src/normalizer.ts

```diff
@@ -360,44 +318,28 @@ class V1SchemaNormalizer {
       if (s.type) {
         return undefined
       }
-      let ref = s.$ref
+
+      // we're ovewriting this string $ref with the link to the
+      // node that we find via query it may or may not have
+      // been overwritten already
       if (typeof s.$ref === 'string') {
-        ref = this.querySchemaRef(s.$ref, [...path, '$ref'])
-        if (ref) {
-          s.$ref = ref
-        }
+        s.$ref = this.querySchemaRef(s.$ref)
       }
 
       s.type = 'object'
```

Comment:
I'm looking at this now and not really sure why this is here. but it's working so i don't want to remove it. will do some more testing and research.